check_stability.py

```diff
@@ -483,31 +482,69 @@ class LogHandler(reader.LogHandler):
         Subclasses reader.LogHandler.
         """
         def __init__(self):
-            self.results = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
+            self.results = []
+
+        def find_or_create_test(self, data):
+            for test in self.results:
+                if test["name"] == data["test"]:
+                    return test
+
+            test = {
+                "name": data["test"],
+                "subtests": [],
+                "status": {"OK": 0, "FAIL": 0}
```

Comment:
This is not the full range of possible statuses. Let's use a defaultdict(int) to avoid depending on specific statuses here.